substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/oldobject/JfrOldObjectTest.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023, 2023, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2023, 2023, Red Hat Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package com.oracle.svm.test.jfr.oldobject;
+
+import com.oracle.svm.core.jfr.JfrEvent;
+import com.oracle.svm.test.jfr.JfrRecordingTest;
+import jdk.jfr.Recording;
+import jdk.jfr.ValueDescriptor;
+import jdk.jfr.consumer.RecordedEvent;
+import jdk.jfr.consumer.RecordedFrame;
+import jdk.jfr.consumer.RecordedObject;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.rules.TestName;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public abstract class JfrOldObjectTest extends JfrRecordingTest {
+    static Object leak;
+
+    @Rule public TestName name = new TestName();
+
+    @Before
+    public void triggerUpdateOfLastKnownHeapUsage() {
+        // Trigger GC before tests to get a reading of last known heap usage for the first executed
+        // test.
+        System.gc();
+        System.gc();
+    }
+
+    @After
+    public void garbageCollectLeak() {
+        // Force previous tests objects to be garbage collected.
+        // The scavenge logic in the sampler should clear those from the queue.
+        leak = null;
+        System.gc();
+        System.gc();
+    }
+
+    Recording startRecording() throws Throwable {
+        final String[] events = {JfrEvent.OldObjectSample.getName()};
+        Map<String, String> settings = new HashMap<>();
+        settings.put("old-objects-stack-trace", "true");
+        return startRecording(events, getDefaultConfiguration(), settings);
+    }
+
+    static void blackhole(Object obj) {
```

Comment:
Good question. The `leak` field is assigned to a static variable but it's not used, so at the beginning when I started working on this I was worried the compiler might optimize this away and remove all the code. The `blackhole` method is just a way to make sure the field is used in a trivial way, but one that cannot be optimized away because it can (extremely unlikely) end up causing a side effect (printing an empty space character).

I've just removed it and calls to it and it all works fine, so I'll go ahead and remove it.